File: 128-longest-consecutive-sequence/longest-consecutive-sequence.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var longestConsecutive = function(nums) {
    const numSet = new Set(nums)

    let maxSequence = 0
    for (const num of nums) {
        if (!numSet.has(num - 1)) {
            let currentNum = num
            let currentSequence = 1

            while (numSet.has(currentNum + 1)) {
                currentNum++
                currentSequence++
            }
            maxSequence = Math.max(currentSequence, maxSequence)
        }
    }
    return maxSequence
};

File: 128-longest-consecutive-sequence/longest-consecutive-sequence.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        nums_set = set(nums)
        best = 0

        for n in nums_set:
            if n - 1 not in nums_set:
                length = 1
                while n + length in nums_set:
                    length += 1
                best = max(best, length)

        return best